File: .NET Core/ToyORB/Messages/NameLookupMessage.cs
using System;
using System.IO;

namespace ToyORB.Messages
{
    public class NameLookupMessage : Message
    {
        public static readonly MessageType MESSAGE_TYPE = MessageType.NameLookup;

        public string ServiceName { get; }

        public NameLookupMessage(string serviceName) : base(MESSAGE_TYPE)
        {
            ServiceName = serviceName ?? throw new ArgumentNullException(nameof(serviceName));
        }

        public override void WriteInto(BinaryWriter stream)
        {
            base.WriteInto(stream);
            stream.WriteBigEndian(ServiceName);
        }

        public override string ToString()
        {
            return "NameLookupMessage{" + ServiceName + "}";
        }

        protected internal class MessageReader : MessageReader<NameLookupMessage>
        {
            public MessageReader() : base(MESSAGE_TYPE) { }

            public override NameLookupMessage ReadMessageFrom(BinaryReader stream)
            {
                return new NameLookupMessage(stream.ReadStringBigEndian());
            }
        }

        protected internal static MessageReader<NameLookupMessage> Reader = new MessageReader();
    }
}

File: Java/src/ToyORB/messages/NameLookupMessage.java
package ToyORB.messages;

import ToyORB.NetUtils;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;

public final class NameLookupMessage extends Message {
	public static final MessageType MESSAGE_TYPE = MessageType.NAME_LOOKUP;

	private final String serviceName;

	public NameLookupMessage(String serviceName) {
		super(MessageType.NAME_LOOKUP);
		this.serviceName = serviceName;
	}

	@Override
	protected void onWriteInto(DataOutputStream stream) throws IOException {
		NetUtils.writeStringInto(stream, getServiceName());
	}

	@Override
	public String toString() {
		return "NameLookupMessage{" + getServiceName() + "}";
	}

	@SuppressWarnings("unused")
	protected static final MessageReader<NameLookupMessage> READER = new MessageReader<NameLookupMessage>(MESSAGE_TYPE) {
		@Override
		public NameLookupMessage readMessageFrom(DataInputStream stream) throws IOException {
			String serviceName = NetUtils.readStringFrom(stream);
			return new NameLookupMessage(serviceName);
		}
	};

	public String getServiceName() {
		return serviceName;
	}
}
